llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -466,6 +468,44 @@ std::string saveModuleProperties(module_split::ModuleDesc &MD,
   return SCFile;
 }
 
+std::string getOptString(module_split::ModuleDesc &MD) {
+  auto &M = MD.getModule();
+  // Process all properties on kernels.
+  for (Function &F : M) {
```

Comment:
What if we have functions with different options metadata in the same module?

Reply:
Module splitting code based on opt level has been added.